app/app/src/main/kotlin/com/hedvig/app/feature/profile/ui/payment/PaymentViewModel2.kt

```diff
@@ -70,39 +75,27 @@ class PaymentViewModel2(
       redeemReferralCodeRepository.redeemReferralCode(code)
         .fold(
           ifLeft = { error -> _uiState.update { it.copy(discountError = error.message) } },
-          ifRight = { data ->
-            _uiState.update {
-              it.copy(
-                activeDiscounts = listOf(
-                  PaymentUiState.Discount(
-                    code = data?.redeemCode?.campaigns?.firstOrNull()?.fragments?.incentiveFragment?.displayValue ?: "",
-                    displayName = data?.redeemCode?.cost?.fragments?.costFragment?.monthlyGross?.fragments?.monetaryAmountFragment?.amount
-                      ?: "",
-                  ),
-                ),
-              )
-            }
-          },
+          ifRight = { data -> loadPaymentData() },
         )
     }
   }
 }
 
 private fun PaymentRepository.PaymentData.toUiState(locale: Locale) = PaymentViewModel2.PaymentUiState(
-  nextChargeAmount = chargeEstimation.format(locale),
+  nextChargeAmount = chargeEstimation?.format(locale),
   nextChargeDate = nextChargeDate,
   insuranceCosts = contracts.map {
     PaymentViewModel2.PaymentUiState.InsuranceCost(
       displayName = it,
-      cost = "X kr/mån",
+      cost = null,
     )
   },
-  totalDiscount = totalDiscount.format(locale),
+  totalDiscount = totalDiscount?.negate()?.format(locale),
   activeDiscounts = redeemedCampagins.map {
-     PaymentViewModel2.PaymentUiState.Discount(
-       code = it.incentive.toString(),
-       displayName = it.displayValue ?: "Unknown"
-     )
+    PaymentViewModel2.PaymentUiState.Discount(
+      code = it.code,
+      displayName = it.displayValue ?: "-",
```

Comment:
To print and format potentially nullable monetary amount values we got this https://github.com/HedvigInsurance/android/blob/90856a690cb8049e9715901c7af7fa9629cb6ee8/app/core/ui-data/src/main/kotlin/com/hedvig/android/core/uidata/UiNullableMoney.kt#L12-L22 is it something that would be useful in this context here?
It also contains this logic of adding "-" if there's no money in there for example.

If not disregard this comment, I am not 100% sure it'd fit here, just thought looks kinda familiar.

Reply:
We can switch to this after we migrate the payment types to octopus. `CurrencyCode` is not present in Giraffe currently which makes this class hard to use without refactoring